packages/material-ui/src/ButtonBase/ButtonBase.js

```diff
@@ -142,10 +142,18 @@ const ButtonBase = React.forwardRef(function ButtonBase(props, ref) {
   const handleTouchStart = useRippleHandler('start', onTouchStart);
   const handleTouchEnd = useRippleHandler('stop', onTouchEnd);
   const handleTouchMove = useRippleHandler('stop', onTouchMove);
+
+  const hadFocusVisibleEventRef = React.useRef(false);
```

Comment:
Are the other components using the focus visible hook impacted by the problem?

Reply:
Good point. I think we should move this logic into `useIsFocusVisible` so that it can't be used incorrectly.